uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/sub-controllers/toil-request-ctrl_test.js

```diff
@@ -0,0 +1,180 @@
+          });
+        });
+
+        it('loads toil amounts', function () {
+          expect(Object.keys($ctrl.toilAmounts).length).toBeGreaterThan(0);
+        });
+
+        it('gets absence types with true sick param', function () {
+          expect(AbsenceTypeAPI.all).toHaveBeenCalledWith({
+            allow_accruals_request: true
+          })
+        });
+
+        it('cannot submit request', function () {
+          expect($ctrl.canSubmit()).toBe(false);
+        });
+
+        describe('with selected duration and dates', function () {
+          beforeEach(function () {
+            var toil_accrue = optionGroupMock.specificObject('hrleaveandabsences_toil_amounts', 'name', 'quarter_day');
+
+            setTestDates(date2016, date2016);
+            $ctrl.request.toilDurationHours = 1;
+            $ctrl.request.updateDuration();
+            $ctrl.request.toil_to_accrue = toil_accrue.value;
+          });
+
+          it('cannot submit request', function () {
+            expect($ctrl.canSubmit()).toBe(true);
+          });
+
+          it('sets expiry date', function() {
+            expect($ctrl.expiryDate).toEqual('2016-07-08');
+          });
+
+          it('calls calculateToilExpiryDate on AbsenceType', function() {
+            expect(AbsenceTypeAPI.calculateToilExpiryDate.calls.mostRecent().args[0]).toEqual('1');
+            expect(AbsenceTypeAPI.calculateToilExpiryDate.calls.mostRecent().args[1]).toEqual('2016-01-12');
+          });
+
+          describe('when user changes number of days selected', function () {
+            beforeEach(function () {
+              $ctrl.changeInNoOfDays();
+            });
+
+            it('resets toil attributes', function () {
+              expect($ctrl.request.toilDurationHours).toEqual(0);
+              expect($ctrl.request.toilDurationMinutes).toEqual(0);
+              expect($ctrl.request.toil_to_accrue).toEqual("");
+            });
+          });
+        });
+      });
+
+      /**
+       * Initialize the controller
+       *
+       * @param leave request
+       */
+      function initTestController(directiveOptions) {
+        $scope = $rootScope.$new();
+
+        $ctrl = $controller('ToilRequestCtrl', {
+          $scope: $scope,
+          $uibModalInstance: modalInstanceSpy,
+          directiveOptions: directiveOptions
+        });
+
+        $scope.$digest();
+      }
+
+      /**
+       * sets from and/or to dates
+       * @param {String} from date set if passed
+       * @param {String} to date set if passed
+       */
+      function setTestDates(from, to) {
+        if (from) {
+          $ctrl.uiOptions.fromDate = new Date(from);
+          $ctrl.updateAbsencePeriodDatesTypes($ctrl.uiOptions.fromDate, 'from');
+          $scope.$digest();
+        }
+
+        if (to) {
+          $ctrl.uiOptions.toDate = new Date(to);
+          $ctrl.updateAbsencePeriodDatesTypes($ctrl.uiOptions.toDate, 'to');
+          $scope.$digest();
```

Comment:
it needs to be called twice otherwise change is only reflected for one of the update call